surge/tasks.py

```diff
@@ -119,6 +126,11 @@ def create_many(cls,
 
         if not all(isinstance(t, dict) for t in tasks_data):
             raise SurgeTaskDataError
+        
+        try:
+            json.dumps(tasks_data)
+        except (TypeError, ValueError):
+            raise SurgeTaskDataError("All task data must be JSON-serializable")
```

Comment:
I think it would be nice to surface the real JSON error too in case it's informative

Reply:
Do you have any preference for how to do that?